woocommerce/wc-admin
# Update task list component to make use of new experimental list

Fixes #6773 

Updated the task-list code to make use of the ExperimentalList that got merged in https://github.com/woocommerce/woocommerce-admin/pull/6787. I also converted the components to functional components.

### Screenshots

<img width="1083" alt="Screen Shot 2021-04-26 at 1 37 58 PM" src="https://user-images.githubusercontent.com/2240960/116119303-a8a9c100-a694-11eb-9de0-fc37de354d51.png">

### Detailed test instructions:

- Start a new WooCommerce store and finish the onboarding flow
- Smoke test the task list, the task list should be have the same way it currently does
     - Hovering over tasks, clicking on tasks, and opening tasks
     - It should trigger the correct tracks (tasklist_view, task_view, tasklist_click)
- Also smoke test the 3PD task list, by following the JS portion of [this tutorial](https://developer.woocommerce.com/2021/02/26/use-setup-tasks-to-provide-a-first-class-on-boarding-experience-for-merchants/), you can create a new extension using `npm run example -- --ext=add-task` 
- Set `isDismissable` to true for one of you tasks and click this, the animation should be correct as well.
